edi_oca/models/edi_exchange_record.py

```diff
@@ -209,7 +211,7 @@ def record(self):
             return None
         if not self.model and self.parent_id:
             return self.parent_id.record
-        return self.env[self.model].browse(self.res_id).exists()
+        return self.env[self.model].sudo(False).search([("id", "=", int(self.res_id))])
```

Comment:
why sudo + false?

Reply:
The initial purpose was to filter out the retrieved record if multi-company rules apply on the model, therefore I suggested the use of `sudo(False).search()` so that the ORM takes care of applying the security rules while executing the `search()`. `sudo(False)` is needed to ensure those rules are actually applied.